ptf_tests/common/utils/test_utils.py

```diff
@@ -639,3 +664,187 @@ def ip_ntns_exec_ping_test(nsname, dst_ip, count="4", remote=False, hostname="",
+    """
+    :Function to check if netperf is installed on linux name space
+    :returns boolean True or False
+    """
+    cmd= "netperf -V"
+    result = gnmi_cli_utis.ip_link_netns_exec(nsname, cmd,remote=remote,hostname=hostname, username=username, password=password)[1]
+    if "Netperf version" not in result:
+        print (f"FAILED: netperf is not installed on namespace {nsname}")
+        return False
+    
+    print (f"PASS: verify netperf is installed on {nsname}")
+    return True
+
+def ipnetns_netperf_client(nsname, host, testlen, testname, option="", remote=False, hostname="",username="",password=""):
+    """
+    :Function to start netperf on linux name space
+    :returns boolean True or False
+    """
+    cmd = f"netperf -H {host} -l {testlen} -t {testname} {option}".replace(u'\xa0', u' ')
+    output = gnmi_cli_utis.ip_link_netns_exec(nsname, cmd,remote=remote,hostname=hostname, username=username, password=password)[1]
+    if testname == "UDP_STREAM":
+        data = output.strip().split("\n")[-1].split()
+        data_len = 4
+    elif testname == "TCP_STREAM":
+        data = output.strip().split("\n")[-1].split()
+        data_len = 5
+    else:
+        print(f"FAIL: No expect netperf execution data collected {output}")
+        return False
+  
+    if len(data) == data_len:
+        for value in data:
+            #all value should be either fload or int
+            if type(float(value)) != float and type(int(value)) == int:
+                print(f"FAIL: Send netperf client failed with {data}")
+                return False    
+    else:
+        print(f"FAIL: Send netperf client failed with {output}")
+        return False
+  
+    print (f"PASS: send netperf {cmd} succeed") 
+    return True
+
+def vm_check_netperf(conn, vmname):
+    """
+    :Function to check of netperf is installed or not
+    :returns boolean True or False
+    """
+    conn.readResult()
+    cmd= "netperf -V"
+    conn.sendCmd(cmd)
+    output = conn.readResult().split("\n")[1]
+    
+    if "Netperf version" not in output:
+        print (f"FAILED: netperf is not installed on {vmname}")
+        return False
+    
+    print (f"PASS: verify netperf is installed on {vmname}")
+    return True
+
+def vm_check_netserver(conn):
+    """
+    To check if a netserver is running or not. If not running, try to start netperf
+    """
+    # read to clean up result
+    conn.readResult()
+    cmd = "ps -ef | grep netserver"
+    conn.sendCmd(cmd)
+    output = conn.readResult().split("\n")
+    if len(output) < 4:
+        net_server_cmd ="netserver" 
+        conn.sendCmd(net_server_cmd)
+        output = conn.readResult().split("\n")
+        if len(output) < 4:
+            print(f"FAIL: Unable to start netserver")
+            return False
+    #print ("PASS: netserver is running on VM") 
+    return True
+
+def vm_netperf_client(conn, host, testlen, testname, option=""):
```

Comment:
output of this util is not very clear. Either format it or print only the values which user is interested into (eg:throughput)

Reply:
Needs to done by Shujun and after will apply in this PR.